cmd/dlt/oidcprovider/cmd.go

```diff
@@ -0,0 +1,197 @@
+/*
+Copyright (c) 2021 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package oidcprovider
+
+import (
+	"fmt"
+	"github.com/openshift/rosa/pkg/aws"
+	"github.com/openshift/rosa/pkg/interactive"
+	"github.com/openshift/rosa/pkg/interactive/confirm"
+	"github.com/openshift/rosa/pkg/logging"
+	"github.com/openshift/rosa/pkg/ocm"
+	rprtr "github.com/openshift/rosa/pkg/reporter"
+	"github.com/spf13/cobra"
+	errors "github.com/zgalor/weberr"
+	"os"
+)
+
+var modes []string = []string{"auto", "manual"}
+
+var args struct {
+	clusterKey string
+	mode       string
+}
+
+var Cmd = &cobra.Command{
+	Use:     "oidc-provider",
+	Aliases: []string{"oidcprovider"},
+	Short:   "Delete OIDC Provider",
+	Long:    "Cleans up OIDC provider of deleted STS cluster.",
+	Example: `  # Delete OIDC provider for cluster named "mycluster"
+  rosa delete oidc-provider --cluster=mycluster`,
+	Run: run,
+}
+
+func init() {
+	flags := Cmd.Flags()
+
+	flags.StringVarP(
+		&args.clusterKey,
+		"cluster",
+		"c",
+		"",
+		"ID of the cluster (deleted/archived) to delete the OIDC provider from (required).",
+	)
+	Cmd.MarkFlagRequired("cluster")
```

Comment:
Just realized that we can query AMS for subscriptions (even for archived clusters) based on cluster name and that will allow us to fetch the cluster ID. For example:
```
$ ocm get /api/accounts_mgmt/v1/subscriptions --parameter=search="display_name='sdq-rosa-dufrn'" | jq -r .items[].cluster_id
1kaquhrti317lh9o0qfedg1dadfqhb76
```
Then we can use the ID to proceed with searching in AWS through tags. We can probably add this as a future improvement, what do you think?